static/src/javascripts/projects/common/modules/identity/consents.js

```diff
@@ -183,38 +160,40 @@ const uncheckAllOptIns = (): Promise<void> =>
 
 const bindUnsubscribeFromAll = (buttonEl: HTMLButtonElement) => {
     buttonEl.addEventListener('click', () => {
-        if (!buttonEl.classList.contains('js-confirm-unsubscribe')) {
-            confirmUnsubscriptionFromAll(buttonEl);
-        } else {
-            addUpdatingState(buttonEl);
-            resetUnsubscribeFromAll(buttonEl);
-            fastdom
-                .read(() => [
-                    ...document.querySelectorAll(
-                        `.${newsletterCheckboxClassName} input:checked`
-                    ),
-                ])
-                .then(checkboxes => {
-                    checkboxes.forEach(inputEl => {
-                        inputEl.checked = false;
-                    });
+        addUpdatingState(buttonEl);
+        resetUnsubscribeFromAll(buttonEl);
+        fastdom
+            .read(() => [
+                ...document.querySelectorAll(
+                    `.${newsletterCheckboxClassName} input:checked`
+                ),
+            ])
+            .then(checkboxes => {
+                checkboxes.forEach(inputEl => {
+                    inputEl.checked = false;
                 });
-            return getCsrfTokenFromElement(
-                document.getElementsByClassName(newsletterCheckboxClassName)[0]
-            )
-                .then(csrfToken => unsubscribeFromAll(csrfToken))
-                .then(() =>
-                    Promise.all([checkAllOptOuts(), uncheckAllOptIns()])
-                )
-                .catch((err: Error) => {
-                    pushError(err, 'reload').then(() => {
-                        window.scrollTo(0, 0);
-                    });
-                })
-                .then(() => {
-                    removeUpdatingState(buttonEl);
+            });
+        return getCsrfTokenFromElement(
+            document.getElementsByClassName(newsletterCheckboxClassName)[0]
+        )
+            .then(csrfToken => unsubscribeFromAll(csrfToken))
+            .then(() => Promise.all([checkAllOptOuts(), uncheckAllOptIns()]))
```

Comment:
I feel setting checkbox state should all be handled in one line, right now it's split between this and the bit on the top